lib/werewolf/engine/state.rb

```diff
@@ -0,0 +1,46 @@
+module Werewolf
+  module Engine
+    class State
+      attr_reader :current, :turn_number, :day_number, :enumerator
+
+      STATES = [:dawn, :day, :dusk, :night]
+      TRANSITIONS = [:dawn, :dusk]
+
+      # We start at turn 0, :dawn on day 0
+      def initialize
+        @enumerator = STATES.cycle
+        @current = @enumerator.next
+        @index, @turn_number, @day_number = 0, 0, 0
```

Comment:
`@index`is extraneous, its just `@turn_number % STATE.length`.  and, on top of that, they need to stay sync'd, but one is part of the public api and one isn't.